nexus/src/nexus.rs

```diff
@@ -1364,7 +1342,7 @@ impl Nexus {
         project_name: &Name,
         instance_name: &Name,
         disk_name: &Name,
-    ) -> DeleteResult {
+    ) -> UpdateResult<Disk> {
```

Comment:
We try to return models rather than views from this layer, and instead convert to view in the route handler. This has the advantage of avoiding the repeated `into`s in here. PRed on this PR in #521